### PR TITLE
:broom: Accommodate to knative-sandbox/reconciler-test#294

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :broom: Accommodate to knative-sandbox/reconciler-test#294

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -84,11 +84,12 @@ require (
...
-	k8s.io/klog v1.0.0 // indirect
 	k8s.io/klog/v2 v2.60.1-0.20220317184644-43cc75f9ae89 // indirect
 	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
 	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace knative.dev/reconciler-test => github.com/cardil/knative-reconciler-test v0.0.0-20220715181934-169264f8ba23
```
[the remark this answers is not in the document]
Comment (cardil) [reply go.mod]: Fixed in cb3a161e04bf49813bf84042002d31b410e3abf8

/unhold

**Comment by cardil** [comment]
Please, review

/cc @knative-sandbox/eventing-writers
/cc @knative-sandbox/control-protocol-approvers

/cc @odacremolbap
/cc @aliok
/cc @matzew
/cc @lionelvillard
/cc @eric-sap
/cc @travis-minke-sap

**Comment by dprotaso** [comment]
looks like this needs a rebase 

I made the `manifest.InstallYamlFS` changes

**Comment by cardil** [comment]
@dprotaso please take a look again

**Comment by dprotaso** [comment]
looks like there's a panic with the e2e stuff

**Comment by cardil** [comment]
Yea. I've also noticed that.

/hold

**Comment by cardil** [comment]
/unhold

Fixed.

@dprotaso Please take a look. With f7f6e07433cb75b8640643edf37488f2284e5077, I've migrated the way to handle test images to the reconciler-test way of things.

File: .ko.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+builds:
+  - id: conformance-go-client
+    main: ./test/test_images/conformance-go-client
+    env:
+      - CGO_ENABLED=0
```
Comment (dprotaso) [inline .ko.yaml:5]: This should be the default with `ko` so I don't think we need to specify it

**Comment by dprotaso** [comment]
/lgtm
/approve
/hold for nit

feel free to unhold

File: .ko.yaml (hunk 2 above)
Comment (cardil) [reply .ko.yaml:5]: Without it, the tests are failing, at least on my machine.

Comment (dprotaso) [reply .ko.yaml:5]: Weird - do you have CGO turn in your environment settings?

https://github.com/google/ko/blob/2f230b88c4891ee3a71b01c1fa65e85e8d6b5f5b/pkg/build/gobuild.go#L381-L386

Comment (cardil) [reply .ko.yaml:5]: No. I don't:

```
$ echo $CGO_ENABLED
$
```

But, I think it's set by default via go:

```
$ go env CGO_ENABLED
1
```

Same, as well, on `docker.io/library/golang` and `quay.io/fedora/fedora:36` images:

```
$ podman run --rm -it docker.io/library/golang go env CGO_ENABLED
1
```

We are running google/ko from within go code, and there the `CGO_ENABLED` is set to `1` by default.

Comment (cardil) [reply .ko.yaml:5]: https://github.com/knative-sandbox/reconciler-test/blob/89e41db3a415a3805e420828d695bf14dcf530e5/pkg/images/ko/publish.go#L35-L44

Comment (dprotaso) [reply .ko.yaml:5]: > We are running google/ko from within go code, and there the CGO_ENABLED is set to 1 by default.

That's very subtle and I would say our invoker of `ko` remove setting `CGO_ENABLED`

**Comment by dprotaso** [comment]
/hold cancel

we should fix how `ko` is invoked - but it shouldn't block this PR

**Pull request merged**